watson_developer_cloud/conversation_v1.py

```diff
@@ -11,108 +13,200 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 """
-The IBM Watson Conversation service combines machine learning, natural
-language understanding, and integrated dialog tools to create conversation
-flows between your apps and your users.
+The IBM Watson Conversation service combines machine learning, natural language
+understanding, and integrated dialog tools to create conversation flows between your apps
+and your users.
 """
 
-from .watson_developer_cloud_service import WatsonDeveloperCloudService
+from __future__ import absolute_import
+
+import json
+from .watson_service import WatsonService
+
+##############################################################################
+# Service
+##############################################################################
 
-class ConversationV1(WatsonDeveloperCloudService):
-    """Client for the Conversation service."""
+
+class ConversationV1(WatsonService):
+    """The Conversation V1 service."""
 
     default_url = 'https://gateway.watsonplatform.net/conversation/api'
-    latest_version = '2017-04-21'
+    VERSION_DATE_2017_05_26 = '2017-05-26'
+    VERSION_DATE_2017_04_21 = '2017-04-21'
+    VERSION_DATE_2017_02_03 = '2017-02-03'
+    VERSION_DATE_2016_09_20 = '2016-09-20'
+    VERSION_DATE_2016_07_11 = '2016-07-11'
+
+    def __init__(self, version, url=default_url, username=None, password=None):
+        """
+        Construct a new client for the Conversation service.
+
+        :param str version: The API version date to use with the service, in
+               "YYYY-MM-DD" format. Whenever the API is changed in a backwards
+               incompatible way, a new minor version of the API is released.
+               The service uses the API version for the date you specify, or
+               the most recent version before that date. Note that you should
+               not programmatically specify the current date at runtime, in
+               case the API has been updated since your application's release.
+               Instead, specify a version date that is compatible with your
+               application, and don't change it until your application is
+               ready for a later version.
 
-    def __init__(self, version, url=default_url, **kwargs):
-        WatsonDeveloperCloudService.__init__(self, 'conversation', url,
-                                             **kwargs)
```

Comment:
Should continue to have the logic of `**kwargs` to support `x_watson_learning_opt_out`, etc.

```    def __init__(self, version, url=default_url, **kwargs):
        WatsonDeveloperCloudService.__init__(self, 'conversation', url,
                                             **kwargs)```

Reply:
You can do `x_watson_learnign_opt_out` with the default headers.  This was recently added.

@ehdsouza We should make sure this is in the migration guide.

Reply:
Also, `**kwargs` is no longer needed for `use_vcap_services`, because it defaults to `True` -- there’s never a case where you want to specify `False`.  Credentials passed in always override vcap services.  If you want vcap services, don’t pass credentials.